faunadb-scala/src/load/scala/faunadb/FaunaClientLoadSpec.scala

```diff
@@ -0,0 +1,76 @@
+package faunadb
+
+import faunadb.query._
+import faunadb.values._
+import org.scalatest.{Matchers, fixture}
+import scala.concurrent.Future
+
+class FaunaClientLoadSpec extends fixture.AsyncWordSpec with Matchers with FaunaClientFixture {
+
+  "When submitting query whose response is about 10 MB of size, it" should {
+    "receive the response with no errors" in { client =>
+      // Fixtures
+      val collectionName = RandomGenerator.aRandomString
+
+      val documentFieldsNumber = 1000
+      val documentValuesSize = 1000
+      val documentsNumber = 10
+
+      val largeDocument: Expr = {
+        val fields: Seq[(String, Expr)] =
+          (1 to documentFieldsNumber).map { _ =>
+            val name = RandomGenerator.aRandomString
+            val value = Expr.encode(RandomGenerator.aRandomString(documentValuesSize))
+            (name -> value)
+          }
+
+        Obj("data" -> Obj(fields: _*))
+      }
+
+      def setup(): Future[(Value, Value)] = {
+        def createCollection(): Future[Value] = client.query(CreateCollection(Obj("name" -> collectionName)))
+
+        def createDocuments(): Future[Seq[Value]] = {
+          def createDocument(): Future[Value] = client.query(Do(Create(Collection(collectionName), largeDocument), Null))
+
+          Future.sequence(
+            (1 to documentsNumber).map { _ =>
+              createDocument()
+            }
+          )
+        }
+
+        val result: Future[(Value, Value)] =
+          for {
+            createCollectionResult <- createCollection()
+            createDocumentsResult <- createDocuments()
+          } yield (createCollectionResult, createDocumentsResult)
+
+        result
+      }
+
+      def run(): Future[Value] = {
+        // Get all documents
+        client.query {
+          Map(
+            Paginate(Documents(Collection(collectionName)), size = documentsNumber),
+            Lambda(ref => Get(ref))
+          )
+        }
+      }
+
+      // Run
+      val result: Future[Value] =
+        for {
+          _ <- setup()
+          result <- run()
+        } yield result
+
+      // Verify
+      result.map(_ => succeed) // If result is returned with no errors, succeed
```

Comment:
TBH I am not sure about the actual size right now :thinking: 
What about adding an assertion to make sure that the payload is actually above 10mb?

Reply:
Yes, that's totally true! I was thinking about adding an assertion on the response size as well, but there's no way to do it without changing the `FaunaClient` API at the moment. In other words, the _size_ value is returned as a Header in the HTTP Response, and that is not exposed outside of the `FaunaClient`. We have this ticket for doing so though: [DRV-250](https://faunadb.atlassian.net/browse/DRV-250).

You might want to debug a query and take a look at the `FullHttpResponse`, when it is returned from the `Connection` here: :)

https://github.com/fauna/faunadb-jvm/blob/fa61aea47bb074db6905df8081129868e1a546d9/faunadb-scala/src/main/scala/faunadb/FaunaClient.scala#L196

In the code, the 10 MB payload comes from these values:

```scala
val documentFieldsNumber = 1000
val documentValuesSize = 1000
val documentsNumber = 10
```

That is, when requesting 10 documents, with 1000 fields, with 1000 chars values, results aprox in a 10 MB response.

I'll add some further comments in the code explaining this!

Reply:
Hooray for comments!